### PR TITLE
Fixed initialization of the master worker thread id

### PR DESCRIPTION
**Pull request by cmnrd**
Fixes https://github.com/lf-lang/reactor-c/issues/452

## Summary by CodeRabbit

- **New Features**
  - Added platform-specific `lf_thread_self()` function for improved thread handling across multiple platforms, including Arduino, FlexPRET, and Windows.

### CODE REVIEW

**Review by edwardalee — APPROVED** [review]
LGTM!  Thanks for fixing this.

**Comment by cmnrd** [comment]
Unfortunately, this does not work as there is no implementation of `lf_thread_self` for most platforms.

File: low_level_platform/impl/src/lf_arduino_support.c [hunk 1]
```diff
@@ -170,6 +170,16 @@ typedef void* (*lf_function_t)(void*);
  */
 int lf_available_cores() { return 1; }
 
+lf_thread_t lf_thread_self() {
+  // Not implemented. Although Arduino mbed provides a ThisThread API and a
+  // get_id() function, it does not provide a way to get the current thread as a
+  // Thread object.
+  // N.B. This wrong implementation will eventually cause hard-to-debug
+  // segfaults, but it unblocks us from conveniently implementing features for
+  // other platforms, and it does not break existing features for Arduino.
```
Comment (cmnrd) [inline low_level_platform/impl/src/lf_arduino_support.c:179]:
```suggestion
  // Not implemented. Although Arduino mbed provides a ThisThread API and a
  // get_id() function, it does not provide a way to get the current thread as a
  // Thread object.
```

File: low_level_platform/impl/src/lf_flexpret_support.c [hunk 2]
```diff
@@ -178,6 +178,13 @@ int lf_available_cores() {
   return FP_THREADS - 1; // Return the number of Flexpret HW threads
 }
 
+lf_thread_t lf_thread_self() {
+  // N.B. This wrong implementation will eventually cause hard-to-debug
+  // segfaults, but it unblocks us from conveniently implementing features for
+  // other platforms, and it does not break existing features for FlexPRET.
```
Comment (cmnrd) [inline low_level_platform/impl/src/lf_flexpret_support.c:184]:
```suggestion
  // Not implemented.
```

**Pull request merged**